tests/test_confidence.py

```diff
@@ -54,15 +54,49 @@ def test_confidence_leastsq(data, pars, verbose, capsys):
         assert 'Calculating CI for' in captured.out
 
 
+def test_confidence_pnames(data, pars):
+    """Test if pnames works as expected."""
+    minimizer = lmfit.Minimizer(residual, pars, fcn_args=(data))
+    out = minimizer.leastsq()
+
+    assert_paramval(out.params['a'], 0.1, tol=0.1)
+    assert_paramval(out.params['b'], -2.0, tol=0.1)
+
+    ci = lmfit.conf_interval(minimizer, out, p_names=['a'])
+
+
+def test_confidence_bounds_reached(data, pars):
+    """Check if conf_interval handles bounds correctly"""
+
+    # Should work
+    pars['a'].max = 0.2
+    minimizer = lmfit.Minimizer(residual, pars, fcn_args=(data))
+    out = minimizer.leastsq()
+    out.params['a'].stderr = 1
+    ci = lmfit.conf_interval(minimizer, out, verbose=True)
+
+    # Should warn
+    pars['b'].max = 2.03
+    pars['b'].min = 1.97
+    minimizer = lmfit.Minimizer(residual, pars, fcn_args=(data))
+    out = minimizer.leastsq()
+    out.params['b'].stderr = 0.005
+    out.params['a'].stderr = 0.01
+    with pytest.warns(UserWarning) as record:
+        ci = lmfit.conf_interval(minimizer, out, verbose=True)
+        assert "Bound reached" in str(record[0].message)
+
```

Comment:
this can be re-written like [or even on two lines, if you use ```......, match=r'Bound reached')```]:
```
msg = r'Bound reached'
with pytest.warns(RuntimeWarning, match=msg):
    lmfit.conf_interval(minimizer, out, verbose=True)
```
and avoids the ```flake8``` warning about unused variable

Reply:
Yeah, the current code is not optimal. I just copied the lines from the other tests.